### DISCUSSION TITLE
What is the best approch for unit conversion

### DISCUSSION THREAD

**Discussion by micker** [dompdf/dompdf#3467] [Q&A]
Hello in order to have the best conversion what is the best approch for unit :
- pixel
- rem
- %
in order to have the best conversion for block size, position etc
thanks for return

**Comment by bsweeney** [comment]
Dompdf's conversion logic is pretty solid, so I'd say more than anything consistency is important. A couple of notes:

- PX: if you want a one-to-one mapping, you'll need to set the Dompdf DPI to 72. This will match the grid density of the PDF (72 px per in = 72 pt per in). Though I have seen people get tripped up using this configuration because the resolution at 72 DPI is relatively small (e.g. for letter-size documents 612x792, minus margins). A higher resolution will naturally give you more space to work with, but you have to ensure you're using consistent units across the document, or the results may not be what you expect.
- REM: this is just a measurement based on the root element's font size. That's resolved to a PT size during computation.

If you want specific control of the placement and size in the PDF then using a real-world measurement (in, mm) or PT is probably your best option.

If you're using any CSS frameworks, I'd check what units are used internally by the framework. One issue that might cause issues for block sizes is the lack of support for the `box-sizing` property.

**Reply by micker** [reply]
thanks for your return !
